graphistry/compute/predicates/comparison.py

```diff
@@ -0,0 +1,354 @@
+from typing import Dict, Union, TYPE_CHECKING
+import pandas as pd
+import numpy as np
+from datetime import datetime, date, time
+
+from .ASTPredicate import ASTPredicate
+from ..ast_temporal import TemporalValue, DateTimeValue, DateValue, TimeValue
+from ...models.gfql.coercions.temporal import to_ast
+from ...models.gfql.types.guards import is_native_numeric, is_any_temporal, is_string
+from ...models.gfql.types.predicates import ComparisonInput, BetweenBoundInput
+from ...utils.json import JSONVal
+from graphistry.compute.typing import SeriesT
+
+if TYPE_CHECKING:
+    pass
+
+
+class ComparisonPredicate(ASTPredicate):
+    """Base class for comparison predicates that support both numeric and temporal values"""
+    
+    def __init__(self, val: ComparisonInput) -> None:
+        self.val = self._normalize_value(val)
+    
+    def _normalize_value(self, val: ComparisonInput) -> Union[int, float, np.number, TemporalValue]:
+        """Convert various input types to internal representation"""
+        # Comparison predicates need:
+        # - Numerics as-is
+        # - Temporals as AST objects (for timezone handling)
+        # - Strings rejected (ambiguous)
+        if is_native_numeric(val):
+            return val
+        elif is_any_temporal(val):
+            return to_ast(val)
+        elif is_string(val):
+            raise ValueError(
+                f"Raw string '{val}' is ambiguous. Use:\n"
+                f"  - pd.Timestamp('{val}') for datetime\n"
+                f"  - {{'type': 'datetime', 'value': '{val}'}} for explicit type"
+            )
+        else:
+            raise TypeError(f"Unsupported type for {self.__class__.__name__}: {type(val)}")
+    
+    def _temporal_comparison(self, s: SeriesT, temporal_val: TemporalValue, op: str) -> SeriesT:
```

Comment:
'op' should be a literal type..

Reply:
also, instead of a big dispatch here, does it make more sense to inline these alongside their numeric counterparts below?